src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.Analyzers/ApiControllerAnalyzer.cs

```diff
@@ -0,0 +1,100 @@
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterSyntaxNodeAction(AnalyzeNode, CodeAnalysis.CSharp.SyntaxKind.SimpleBaseType);
+            context.RegisterSyntaxNodeAction(AnalyzeNode, CodeAnalysis.VisualBasic.SyntaxKind.InheritsStatement);
+        }
+
+        private void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var namedTypeSymbol = context.ContainingSymbol as INamedTypeSymbol;
+
+            if (namedTypeSymbol != null && namedTypeSymbol.BaseType != null
+                && namedTypeSymbol.BaseType.ToDisplayString().Equals($"{BadNamespace}.{BadClassName}", StringComparison.Ordinal))
+            {
+                if (context.Compilation.Language == LanguageNames.CSharp)
+                {
+                    ReportCSharpSyntax(context);
+                }
+                else if (context.Compilation.Language == LanguageNames.VisualBasic)
+                {
+                    ReportVisualBasicSyntax(context);
+                }
+            }
+        }
+
+        private static void ReportCSharpSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var diagnostic = Diagnostic.Create(Rule, context.Node.GetLocation(), context.Node.ToString());
+            context.ReportDiagnostic(diagnostic);
+        }
+
+        private static void ReportVisualBasicSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var baseClass = context.Node.DescendantNodes()
+                .OfType<CodeAnalysis.VisualBasic.Syntax.QualifiedNameSyntax>()
+                .FirstOrDefault() as SyntaxNode;
+
+            if (baseClass is null)
+            {
+                baseClass = context.Node.DescendantNodes()
+                .OfType<CodeAnalysis.VisualBasic.Syntax.IdentifierNameSyntax>()
+                .FirstOrDefault();
+            }
+
+            var diagnostic = Diagnostic.Create(Rule, baseClass.GetLocation(), baseClass.ToString());
+            context.ReportDiagnostic(diagnostic);
+        }
+
+        public static Project AddMetadataReferences(Project project)
+        {
+            if (project is null)
+            {
+                return project!;
+            }
+
+            // todo - still an open question about how we locate metadatareferences
+            const string assemblyFolder = @"C:\deleteMe\";
+            var assemblyPath = Path.Combine(assemblyFolder, $"System.Web.Http.dll");
```

Comment:
This assembly comes from NuGet while others could come from locally installed .NET framework

Reply:
Let's chat about this tomorrow. There's a reference assembly nupkg that we can grab